chainer/functions/noise/gumbel_softmax.py

```diff
@@ -0,0 +1,40 @@
+from chainer import cuda
+from chainer.functions.activation import softmax
```

Comment:
Use `chainer.functions.softmax` instead of `softmax.softmax` (it is allowed inside of any functions).